[Merged by Bors] - feat(Algebra/GroupWithZero.Commute): add Ring.inverse_pow_mul_eq_iff_eq_mul

mariainesdff opened the pull request:
Co-authored-by: AntoineChambert-Loir <antoine.chambert-loir@u-paris.fr>


---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

j-loreaux commented on Mathlib/Algebra/GroupWithZero/Commute.lean:
```diff
@@ -38,6 +38,11 @@ lemma inverse_pow (r : M₀) : ∀ n : ℕ, Ring.inverse r ^ n = Ring.inverse (r
+    (ha : IsUnit a) {k : ℕ} : Ring.inverse a ^ k * b = c ↔ b = a ^ k * c := by
+  rw [Ring.inverse_pow, Ring.inverse_mul_eq_iff_eq_mul _ _ _ (IsUnit.pow _ ha)]
+
+
```
```suggestion
```

j-loreaux reviewed (commented):
To me, this seems to be right on the edge of being useful.

maintainer delegate

riccardobrasca reviewed (approved):
`Ring.inverse` is not about rings :D

Anyway, what about calling the lemma `IsUnit.something` to allow dot notation?

bors d+

mariainesdff commented:
> Anyway, what about calling the lemma `IsUnit.something` to allow dot notation?
I have renamed it `IsUnit.ring_inverse_pow_mul_eq_iff_eq_mul` (and I dropped the commutativity hypothesis).

eric-wieser commented:
> Anyway, what about calling the lemma `IsUnit.something` to allow dot notation?

This makes it inconsistent with `Ring.inverse_mul_eq_iff_eq_mul`, which is otherwise a very similar lemma. I have a slight preference for the original name, and a strong preference that all the basic lemmas about `Ring.inverse` and `IsUnit` are namespaced consistently.

riccardobrasca commented:
> > Anyway, what about calling the lemma `IsUnit.something` to allow dot notation?
> 
> This makes it inconsistent with `Ring.inverse_mul_eq_iff_eq_mul`, which is otherwise a very similar lemma. I have a slight preference for the original name, and a strong preference that all the basic lemmas about `Ring.inverse` and `IsUnit` are namespaced consistently one way or the other.

I agree we want to be consistent. We can always change the name later.

eric-wieser commented:
So let's revert ded5e49ab53eb48a4dcaeee3b6720a883c2546e2 then?

mariainesdff commented:
bors r+
